src/compiler/watchPublic.ts

```diff
@@ -434,15 +435,22 @@ namespace ts {
             const hasInvalidatedResolution = resolutionCache.createHasInvalidatedResolution(userProvidedResolution || changesAffectResolution);
             if (isProgramUptoDate(getCurrentProgram(), rootFileNames, compilerOptions, getSourceVersion, fileExists, hasInvalidatedResolution, hasChangedAutomaticTypeDirectiveNames, getParsedCommandLine, projectReferences)) {
                 if (hasChangedConfigFileParsingErrors) {
+                    if (reportWatchDiagnosticOnCreateProgram) {

```

Comment:
Seems like this has to be true?  How could the config file change without triggering a program refresh?

Reply:
Unless I'm mistaken, `hasChangedConfigFileParsingErrors` can be true if `hasChangedConfigFileParsingErrors` was called, which can happen if the refresh level was `ConfigFileProgramReloadLevel.Partial` (and then those call sites are outside of the watcher changes, entirely, I think). I'll remove the if and see if there are any tests for this, though.

Reply:
It seems like that if there's a case where `hasChangedConfigFileParsingErrors` is true but `reportWatchDiagnosticOnCreateProgram` is false, there's no test for it.

I can change this to unconditionally report the message, but I'm not entirely confident that there's no code path where this breaks.

Reply:
I'm not familiar from this code - I'm just pointing out that, abstractly, it seems like a config file change will always be a relevant file change (i.e. will require the message to be printed).